src/shared/watchdog.c

```diff
@@ -195,10 +195,13 @@ usec_t watchdog_get_last_ping(clockid_t clock) {
 }
 
 static int watchdog_ping_now(void) {
+        int r;
+
         assert(watchdog_fd >= 0);
 
-        if (ioctl(watchdog_fd, WDIOC_KEEPALIVE, 0) < 0)
-                return log_warning_errno(errno, "Failed to ping hardware watchdog, ignoring: %m");
+        r = RET_NERRNO(ioctl(watchdog_fd, WDIOC_KEEPALIVE, 0));
+        if (r < 0)
+                return log_warning_errno(r, "Failed to ping hardware watchdog, ignoring: %m");
```

Comment:
btw, any chance you can add ratelimiting to this message too? in order to address #35405. as it appears there are broken watchdogs around that can be opened but where pinging them fails. we should probably warn about that a couple of times, but then give up. 

(also, we might want to consider closing the fd here again on failure. (either here or in watchdog_ping() or watchdog_open() after calling watchdog_ping_now()). but that's another story...)

Reply:
Yeah, I think we should close the fd. If the watchdog works, and we don't close it, we'd get rebooted. So we should at least try to close it.